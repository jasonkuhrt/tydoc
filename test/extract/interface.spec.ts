it('extracts doc from interface', () => {
  expect(
    ctx.given(`
      export interface A {
        foo: string
        bar: string
      }
    `)
  ).toMatchSnapshot()
})

it('if exported with as-alias then export name reflects alias name', () => {
  expect(
    ctx.given(`
      interface A {}
      export { A as B }
    `)
  ).toMatchSnapshot()
})

it('export as-alias names do not conflict with type names in the type index', () => {
  expect(
    ctx.given(`
      interface A {
        c: C
      }

      interface C {}

      export { A as C }
    `)
  ).toMatchSnapshot()
})
