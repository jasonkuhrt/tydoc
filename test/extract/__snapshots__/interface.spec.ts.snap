// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extracts doc from interface 1`] = `
Object {
  "hybrids": Array [],
  "length": 1,
  "terms": Array [],
  "types": Array [
    Object {
      "jsDoc": null,
      "kind": "interface",
      "languageLevel": "type",
      "name": "A",
      "properties": Array [
        Object {
          "jsDoc": null,
          "name": "foo",
          "type": Object {
            "isPrimitive": true,
            "name": "string",
          },
        },
        Object {
          "jsDoc": null,
          "name": "bar",
          "type": Object {
            "isPrimitive": true,
            "name": "string",
          },
        },
      ],
      "sourceLocation": Object {
        "fileLine": 1,
        "filePath": "/test.ts",
      },
      "text": "export interface A {
  foo: string;
  bar: string;
}",
    },
  ],
}
`;
